### PR TITLE
Some improvements to error messages

### PR DESCRIPTION
**Pull request by goshakkk**
@lydell in https://github.com/brunch/brunch/issues/1294#issuecomment-214164581 suggested a few ideas on improving our error messages, to make them friendlier and more actionable.

- Warn when there is nothing to compile
- Friendlier and more suggestive config and package.json errors

@paulmillr:

- [ ] merge https://github.com/paulmillr/loggy/pull/13
- [ ] release loggy 0.3.3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "fcache": "~0.1.0",
     "init-skeleton": "~0.9.0",
     "skemata": "~0.1.0",
-    "loggy": "~0.3.0",
+    "loggy": "~0.3.3",
```
Comment (goshakkk) [inline package.json]: the loggy PR **has** to be merged & loggy released before this is merged

File: lib/config.js [hunk 2]
```diff
@@ -560,7 +560,13 @@ const loadNpm = (config) => {
       try {
         json = require(jsonPath);
       } catch (error) {
-        return reject(new Error('You probably need to execute `npm install` to install brunch plugins. ' + error));
+        if (error.message.indexOf('Cannot find module') !== -1) {
+          return reject(new Error(`No "package.json" was found. Brunch is not very useful without plugins. Run \`npm init\` and add "javascript-brunch" and "css-brunch" to get started to your "package.json" dependencies. Oh, and don't forget to \`npm install\` after that!`));
```
Comment (lydell) [inline lib/config.js]: What about this?

```
No "package.json" was found. That's where brunch plugins need to be listed. Plugins tell Brunch how to compile your files, so without any plugins Brunch won't know what to do. To get started, run `npm init` to generate a "package.json" file and `npm install --save javascript-brunch css-brunch` to install and save plugins for regular JS and CSS.
```

Comment (goshakkk) [inline lib/config.js]: :+1:

File: lib/config.js [hunk 3]
```diff
@@ -616,8 +631,8 @@ const tryToLoad = (configPath, fallbackHandler) => {
     if (!config) {
       return Promise.reject(new Error(`${basename} must be a valid object`));
     }
-    if (!config.files) {
-      return Promise.reject(new Error(`${basename} must have "files" property`));
+    if (!('files' in config)) {
+      return Promise.reject(new Error(`${basename} must have "files" property. Here's a minimal brunch-config.js to get you started:\n${minimalConfig}`));
```
Comment (lydell) [inline lib/config.js]: Above, `"package.json"` was written with quotes, but here `brunch-config.js` (and possibly `${basename}`?) is written without quotes. I know, this is really nitpicky, but perhaps it's cleanest to consistently drop the quotes?

File: lib/config.js [hunk 4]
```diff
@@ -626,7 +641,8 @@ const tryToLoad = (configPath, fallbackHandler) => {
       if (!fallbackHandler) {
         fallbackHandler = () => {
           logger.error(`The directory does not seem to be a Brunch project. Create ${basename}.js or run brunch from the correct directory. ${error.toString().replace('Error: ', '')}`);
-          process.exit(0);
+          logger.error(`Here's a minimal brunch-config.js to get you started:\n${minimalConfig}`);
+          process.exit(1);
```
Comment (lydell) [inline lib/config.js]: Same here :)

File: lib/config.js (hunk 2 above)
Comment (goshakkk) [inline lib/config.js]: Yeah, let's drop these quotes

File: lib/config.js [hunk 5]
```diff
@@ -560,7 +560,13 @@ const loadNpm = (config) => {
       try {
         json = require(jsonPath);
       } catch (error) {
-        return reject(new Error('You probably need to execute `npm install` to install brunch plugins. ' + error));
+        if (error.message.indexOf('Cannot find module') !== -1) {
```
Comment (lydell) [inline lib/config.js]:
```
$ node
> try { require('./mash') } catch (error) { console.log(error) }
{ [Error: Cannot find module './mash'] code: 'MODULE_NOT_FOUND' }
```

Perhaps checking for `error.code === 'MODULE_NOT_FOUND` is more reliable?

File: lib/config.js [hunk 6]
```diff
@@ -560,7 +560,13 @@ const loadNpm = (config) => {
...
+          return reject(new Error(`No "package.json" was found. Brunch is not very useful without plugins. Run \`npm init\` and add "javascript-brunch" and "css-brunch" to get started to your "package.json" dependencies. Oh, and don't forget to \`npm install\` after that!`));
+        } else if (error.message.indexOf('SyntaxError:') !== -1 || error.message.indexOf('Unexpected end of input') !== -1) {
```
Comment (lydell) [inline lib/config.js]:
```
$ node
> try { require('./package.json') } catch (error) { console.log(error instanceof SyntaxError) }
true
```

`error instanceof SyntaxError` might be a cleaner check.

File: lib/config.js (hunk 5 above)
Comment (lydell) [inline lib/config.js]: GitHub marked the above as a comment on an outdated diff for some reason.

File: lib/config.js (hunk 6 above)
Comment (lydell) [inline lib/config.js]: GitHub marked the above as a comment on an outdated diff for some reason.

**Pull request merged**